nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -197,11 +202,14 @@ in
         }
       ];
 
-      systemd.services.lemmy = {
+      systemd.services.lemmy = let
+        configFile = "${settingsFormat.generate "config.hjson" cfg.settings}";
+        mergedConfig = "/tmp/lemmy-config.hjson";
```

Comment:
This is OK, but I think using runtime directory to store the merged config is better because it allows users with enough privilege, such as root, to check the content of that file.

To use runtime directory, `RuntimeDirectory` and `RuntimeDirectoryMode` needs to be set.  Check the [dendrite][dendrite] module for an example.

[dendrite]: https://github.com/NixOS/nixpkgs/blob/1c9db9710cb23d60570ad4d7ab829c2d34403de3/nixos/modules/services/matrix/dendrite.nix#L296

Reply:
I think the default `RuntimeDirectoryMode=0755` is fine since it protects the directory from being modified by other users.
Reading the directory is no problem since the config is only readable by the service, I created tests to ensure that.